### ISSUE TITLE
Any Chance for Help with LG UltraWide?

### ISSUE THREAD

**Issue by kenrucker** [Floris497/mac-pixel-clock-patch#11]
I tried. I really did. Got the the asm things and got very lost.

Does this help? You're way smarter than me! Yikes!

md5 -q /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit

Returns

131978134faf623c7803458c2a204d60

...and...

_CheckTimingWithRange:
000000000001a298	pushq	%rbp
000000000001a299	movq	%rsp, %rbp
000000000001a29c	pushq	%r15
000000000001a29e	pushq	%r14
000000000001a2a0	pushq	%r13
000000000001a2a2	pushq	%r12
000000000001a2a4	pushq	%rbx
000000000001a2a5	movl	0x20(%rsi), %ecx
000000000001a2a8	movl	$0x1, %eax
000000000001a2ad	testb	$0x1, %cl
000000000001a2b0	jne	0x1a648
000000000001a2b6	testb	$0x4, %cl
000000000001a2b9	je	0x1a2ca
000000000001a2bb	movl	$0x22, %eax
000000000001a2c0	testb	$0xc, 0x3c(%rdi)
000000000001a2c4	je	0x1a648
000000000001a2ca	movl	0x80(%rsi), %edx
000000000001a2d0	cmpl	$0x2, %edx
000000000001a2d3	jb	0x1a2ec
000000000001a2d5	movl	0xd0(%rdi), %ebx
000000000001a2db	testl	%ebx, %ebx
000000000001a2dd	je	0x1a2ec
000000000001a2df	movl	$0x23, %eax
000000000001a2e4	cmpl	%ebx, %edx
000000000001a2e6	ja	0x1a648
000000000001a2ec	movl	0x40(%rsi), %r15d
000000000001a2f0	movl	0x44(%rsi), %r14d
000000000001a2f4	movl	0x50(%rsi), %r10d
000000000001a2f8	movl	0x54(%rsi), %r9d
000000000001a2fc	movq	%r14, %r11
000000000001a2ff	addq	%r15, %r11
000000000001a302	movl	$0x24, %eax
000000000001a307	je	0x1a648
000000000001a30d	movq	%r9, %r8
000000000001a310	addq	%r10, %r8
000000000001a313	je	0x1a648
000000000001a319	movq	0x28(%rsi), %rbx
000000000001a31d	movl	$0x4, %eax
000000000001a322	cmpq	0x28(%rdi), %rbx
000000000001a326	ja	0x1a648
000000000001a32c	cmpq	0x20(%rdi), %rbx
000000000001a330	jb	0x1a648
000000000001a336	xorl	%edx, %edx
000000000001a338	movq	%rbx, %rax
000000000001a33b	divq	%r11
000000000001a33e	movq	%rax, %r13
000000000001a341	movl	0x4c(%rdi), %r12d
000000000001a345	movl	$0x5, %eax
000000000001a34a	cmpq	%r12, %r13
000000000001a34d	ja	0x1a648
000000000001a353	movl	0x48(%rdi), %edx
000000000001a356	cmpq	%rdx, %r13
000000000001a359	jb	0x1a648
000000000001a35f	shrb	$0x2, %cl
000000000001a362	andb	$0x1, %cl
000000000001a365	shlq	%cl, %rbx
000000000001a368	movq	%r8, %rcx
000000000001a36b	imulq	%r11, %rcx
000000000001a36f	xorl	%edx, %edx
000000000001a371	movq	%rbx, %rax
000000000001a374	divq	%rcx
000000000001a377	movq	%rax, %rcx
000000000001a37a	movl	0x44(%rdi), %edx
000000000001a37d	movl	$0x6, %eax
000000000001a382	cmpq	%rdx, %rcx
000000000001a385	ja	0x1a648
000000000001a38b	movl	0x40(%rdi), %edx
000000000001a38e	cmpq	%rdx, %rcx
000000000001a391	jb	0x1a648
000000000001a397	movl	0x50(%rdi), %ecx
000000000001a39a	movl	$0x7, %eax
000000000001a39f	cmpq	%rcx, %r11
000000000001a3a2	ja	0x1a648
000000000001a3a8	movl	0x54(%rdi), %ecx
000000000001a3ab	movl	$0x8, %eax
000000000001a3b0	cmpq	%rcx, %r8
000000000001a3b3	ja	0x1a648
000000000001a3b9	movl	$0x9, %eax
000000000001a3be	cmpl	0x74(%rdi), %r15d
000000000001a3c2	ja	0x1a648
000000000001a3c8	cmpl	0x70(%rdi), %r15d
000000000001a3cc	jb	0x1a648
000000000001a3d2	movl	$0xa, %eax
000000000001a3d7	cmpl	0x94(%rdi), %r10d
000000000001a3de	ja	0x1a648
000000000001a3e4	cmpl	0x90(%rdi), %r10d
000000000001a3eb	jb	0x1a648
000000000001a3f1	movl	0x48(%rsi), %r13d
000000000001a3f5	movl	$0xd, %eax
000000000001a3fa	cmpl	0x84(%rdi), %r13d
000000000001a401	ja	0x1a648
000000000001a407	cmpl	0x80(%rdi), %r13d
000000000001a40e	jb	0x1a648
000000000001a414	movl	0x4c(%rsi), %r12d
000000000001a418	movl	$0xe, %eax
000000000001a41d	cmpl	0x8c(%rdi), %r12d
000000000001a424	ja	0x1a648
000000000001a42a	cmpl	0x88(%rdi), %r12d
000000000001a431	jb	0x1a648
000000000001a437	movl	0x58(%rsi), %ebx
000000000001a43a	movl	$0xf, %eax
000000000001a43f	cmpl	0xa4(%rdi), %ebx
000000000001a445	ja	0x1a648
000000000001a44b	cmpl	0xa0(%rdi), %ebx
000000000001a451	jb	0x1a648
000000000001a457	movl	0x5c(%rsi), %ecx
000000000001a45a	movl	$0x10, %eax
000000000001a45f	cmpl	0xac(%rdi), %ecx
000000000001a465	ja	0x1a648
000000000001a46b	cmpl	0xa8(%rdi), %ecx
000000000001a471	jb	0x1a648
000000000001a477	movl	0x60(%rsi), %edx
000000000001a47a	movl	%edx, -0x2c(%rbp)
000000000001a47d	movl	$0x11, %eax
000000000001a482	cmpl	0xb4(%rdi), %edx
000000000001a488	ja	0x1a648
000000000001a48e	movl	-0x2c(%rbp), %edx
000000000001a491	cmpl	0xb0(%rdi), %edx
000000000001a497	jb	0x1a648
000000000001a49d	movl	0x64(%rsi), %edx
000000000001a4a0	movl	%edx, -0x30(%rbp)
000000000001a4a3	movl	$0x12, %eax
000000000001a4a8	cmpl	0xbc(%rdi), %edx
000000000001a4ae	ja	0x1a648
000000000001a4b4	movl	-0x30(%rbp), %edx
000000000001a4b7	cmpl	0xb8(%rdi), %edx
000000000001a4bd	jb	0x1a648
000000000001a4c3	movl	0x68(%rsi), %edx
000000000001a4c6	movl	%edx, -0x34(%rbp)
000000000001a4c9	movl	$0x13, %eax
000000000001a4ce	cmpl	0xc4(%rdi), %edx
000000000001a4d4	ja	0x1a648
000000000001a4da	movl	-0x34(%rbp), %edx
000000000001a4dd	cmpl	0xc0(%rdi), %edx
000000000001a4e3	jb	0x1a648
000000000001a4e9	movl	0x6c(%rsi), %edx
000000000001a4ec	movl	%edx, -0x38(%rbp)
000000000001a4ef	movl	$0x14, %eax
000000000001a4f4	cmpl	0xcc(%rdi), %edx
000000000001a4fa	ja	0x1a648
000000000001a500	movl	-0x38(%rbp), %edx
000000000001a503	cmpl	0xc8(%rdi), %edx
000000000001a509	jb	0x1a648
000000000001a50f	movzbl	0x60(%rdi), %esi
000000000001a513	xorl	%edx, %edx
000000000001a515	movl	%r15d, %eax
000000000001a518	divl	%esi
000000000001a51a	movl	$0x15, %eax
000000000001a51f	testl	%edx, %edx
000000000001a521	jne	0x1a648
000000000001a527	movzbl	0x61(%rdi), %esi
000000000001a52b	xorl	%edx, %edx
000000000001a52d	movl	%r14d, %eax
000000000001a530	divl	%esi
000000000001a532	movl	$0x16, %eax
000000000001a537	testl	%edx, %edx
000000000001a539	jne	0x1a648
000000000001a53f	movzbl	0x62(%rdi), %esi
000000000001a543	xorl	%edx, %edx
000000000001a545	movl	%r13d, %eax
000000000001a548	divl	%esi
000000000001a54a	movl	$0x17, %eax
000000000001a54f	testl	%edx, %edx
000000000001a551	jne	0x1a648
000000000001a557	movzbl	0x63(%rdi), %esi
000000000001a55b	xorl	%edx, %edx
000000000001a55d	movl	%r12d, %eax
000000000001a560	divl	%esi
000000000001a562	movl	$0x18, %eax
000000000001a567	testl	%edx, %edx
000000000001a569	jne	0x1a648
000000000001a56f	movzbl	0x64(%rdi), %esi
000000000001a573	xorl	%edx, %edx
000000000001a575	movl	%r10d, %eax
000000000001a578	divl	%esi
000000000001a57a	movl	$0x22, %eax
000000000001a57f	testl	%edx, %edx
000000000001a581	jne	0x1a648
000000000001a587	movzbl	0x65(%rdi), %esi
000000000001a58b	xorl	%edx, %edx
000000000001a58d	movl	%r9d, %eax
000000000001a590	divl	%esi
000000000001a592	movl	$0x19, %eax
000000000001a597	testl	%edx, %edx
000000000001a599	jne	0x1a648
000000000001a59f	movzbl	0x66(%rdi), %esi
000000000001a5a3	xorl	%edx, %edx
000000000001a5a5	movl	%ebx, %eax
000000000001a5a7	divl	%esi
000000000001a5a9	movl	$0x1a, %eax
000000000001a5ae	testl	%edx, %edx
000000000001a5b0	jne	0x1a648
000000000001a5b6	movzbl	0x67(%rdi), %esi
000000000001a5ba	xorl	%edx, %edx
000000000001a5bc	movl	%ecx, %eax
000000000001a5be	divl	%esi
000000000001a5c0	movl	$0x1b, %eax
000000000001a5c5	testl	%edx, %edx
000000000001a5c7	jne	0x1a648
000000000001a5c9	movzbl	0x68(%rdi), %ecx
000000000001a5cd	xorl	%edx, %edx
000000000001a5cf	movl	-0x2c(%rbp), %eax
000000000001a5d2	divl	%ecx
000000000001a5d4	movl	$0x1c, %eax
000000000001a5d9	testl	%edx, %edx
000000000001a5db	jne	0x1a648
000000000001a5dd	movzbl	0x69(%rdi), %ecx
000000000001a5e1	xorl	%edx, %edx
000000000001a5e3	movl	-0x30(%rbp), %eax
000000000001a5e6	divl	%ecx
000000000001a5e8	movl	$0x1d, %eax
000000000001a5ed	testl	%edx, %edx
000000000001a5ef	jne	0x1a648
000000000001a5f1	movzbl	0x6a(%rdi), %ecx
000000000001a5f5	xorl	%edx, %edx
000000000001a5f7	movl	-0x34(%rbp), %eax
000000000001a5fa	divl	%ecx
000000000001a5fc	movl	$0x1e, %eax
000000000001a601	testl	%edx, %edx
000000000001a603	jne	0x1a648
000000000001a605	movzbl	0x6b(%rdi), %ecx
000000000001a609	xorl	%edx, %edx
000000000001a60b	movl	-0x38(%rbp), %eax
000000000001a60e	divl	%ecx
000000000001a610	movl	$0x1f, %eax
000000000001a615	testl	%edx, %edx
000000000001a617	jne	0x1a648
000000000001a619	movzbl	0x6c(%rdi), %ecx
000000000001a61d	xorl	%edx, %edx
000000000001a61f	movq	%r11, %rax
000000000001a622	divq	%rcx
000000000001a625	movl	$0x20, %eax
000000000001a62a	testq	%rdx, %rdx
000000000001a62d	jne	0x1a648
000000000001a62f	movzbl	0x6d(%rdi), %ecx
000000000001a633	xorl	%esi, %esi
000000000001a635	xorl	%edx, %edx
000000000001a637	movq	%r8, %rax
000000000001a63a	divq	%rcx
000000000001a63d	testq	%rdx, %rdx
000000000001a640	movl	$0x21, %eax
000000000001a645	cmovel	%esi, %eax
000000000001a648	popq	%rbx
000000000001a649	popq	%r12
000000000001a64b	popq	%r13
000000000001a64d	popq	%r14
000000000001a64f	popq	%r15
000000000001a651	popq	%rbp
000000000001a652	retq

**Comment by kenrucker** [comment]
P.S. This is for El Capitan.

**Comment by Tricheur-one** [comment]
I can confirm the md5 hash is 131978134faf623c7803458c2a204d60 for IOKit on El Capitan.

Unable to deal with ASM too :(

**Comment by Floris497** [comment]
Whait, will look into that in a minute!

**Comment by Floris497** [comment]
Added that md5 hash, seemed like my Mac secretly updated to 10.11.1

**Comment by kenrucker** [comment]
Outstanding. What we do without people who understand this stuff. Will give it a try. Thanks!

**Comment by Floris497** [comment]
Actually it is very simple :D I didn't do anything with assembly before this, but basically your let the function jump right to the end instead if letting it perfom all the checks. And as far as the AMD and Nvidia files, you just replace 165000 with 400000 :D

**Comment by kenrucker** [comment]
Hmm. Am I doing something wrong? Is there some type of permission I should set?

Ken-iMac:Downloads Ken$ ./macPixelClockPatcher.command
Detected unpatched IOKit on 10.11, patching.
Can't rename /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit to /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.bak: Operation not permitted, skipping file.
/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit: replacing existing signature
/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit: Operation not permitted
Detected unpatched NVIDIA driver on 10.11, patching.
Can't rename /System/Library/Extensions/NVDAGK100Hal.kext/Contents/MacOS/NVDAGK100Hal to /System/Library/Extensions/NVDAGK100Hal.kext/Contents/MacOS/NVDAGK100Hal.bak: Operation not permitted, skipping file.
Ken-iMac:Downloads Ken$

**Comment by Floris497** [comment]
Yes! System Integrity protection must be disabled, in 10.11 even the root user has no access to those files.  Can be disabled from the recovery partition.

**Comment by Tricheur-one** [comment]
It works fine! Thank you!

**Comment by kenrucker** [comment]
Awesome. Works. Thank you for spending your time helping with this!

For anyone else who needs this, such as a PC guy like me, here is what I did.

Rebooted into Recovery mode by holding down COMMAND and R as the Mac rebooted, until the Apple logo appeared. From Recovery Mode, clicked the Utilities menu at the top and selected Terminal.

In Terminal I typed "csrutil disable".

"csrutil enable --without debug" was mentioned by another site but did not work.

Rebooted. Re-ran Floris's command. Rebooted, and it worked. I then did another reboot into Recovery Mode, opened Terminal and re-enabled system protection via "csrutil enable", rebooted, and it still works fine. I did this last step just to keep the Mac secure since the command script has already done it's work I figured no need to leave the system wide open.

Thanks again!
